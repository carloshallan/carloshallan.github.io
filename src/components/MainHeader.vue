<template>
  <header>
    <div class="logo">
      <main-logo class="svg" />
      <span>
        ${<span class="c">C</span>.<span class="Hallan">Hallan</span>}
      </span>
    </div>
    <navigator />
    <nav class="languages">
      <span
        v-for="(locale, index) in languages"
        :key="index"
        :class="{ active: currentLanguage === locale }"
        @click="changeLanguage(locale)"
      >
        {{ locale }}
      </span>
    </nav>
  </header>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue'
import MainLogo from '@/components/icons/MainLogo.vue'
import Navigator from '@/components/Navigator.vue'

export default defineComponent({
  name: 'MainHeader',
  components: { MainLogo, Navigator },
  setup() {
    let currentLanguage = ref('EN')
    const languages: string[] = ['EN', 'PT-BR']

    return {
      languages,
      currentLanguage,
    }
  },
  methods: {
    changeLanguage(locale: string) {
      this.currentLanguage = locale
      console.log(locale, this.currentLanguage)
    },
  },
})
</script>

<style lang="stylus" scoped>
@import "../styles/default"

header
  width 100vw
  height auto
  position fixed
  top 0
  left: 0
  display flex
  align-items center
  justify-content space-between
  padding: 15px
  background-color dark

header
  .logo
    display flex
    align-items center
    justify-content left
    color white

  .logo .c
    color blue

  .logo .Hallan
    color light-pink

  .svg
    margin-right 10px
    width 40px
    height auto

  .languages
    color white

    span:not(:last-child)::after
     content "|"
     padding 0 10px

    span.active
     color light-pink
</style>
