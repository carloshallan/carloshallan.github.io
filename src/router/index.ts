import Vue from 'vue'
import VueRouter, { RouteConfig } from 'vue-router'

Vue.use(VueRouter)

const HomeComponent = () => import('@/views/Home.vue')
const WorkComponent = () => import('@/views/Work.vue')
const PostComponent = () => import('@/views/PostView.vue')
const PageNotFound = () => import('@/views/PageNotFound.vue')

const routes: Array<RouteConfig> = [
  {
    path: '/',
    name: 'Home',
    component: HomeComponent
  },
  {
    path: '/work',
    name: 'Work',
    component: WorkComponent
  },
  {
    path: '/posts/:slug',
    name: 'post',
    component: PostComponent
  },
  {
    path: '*',
    name: 'PageNotFound',
    component: PageNotFound
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes
})

export default router
