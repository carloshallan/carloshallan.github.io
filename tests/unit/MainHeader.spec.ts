import { mount } from '@vue/test-utils'
import MainHeader from '@/components/MainHeader.vue'
import router from '@/router'

describe('MainHeader.vue', () => {
  it('Render MainHeader', () => {
    const mainHeader = mount(MainHeader, { global: { plugins: [router] } })
    expect(mainHeader.element).toMatchSnapshot()
  })
})
