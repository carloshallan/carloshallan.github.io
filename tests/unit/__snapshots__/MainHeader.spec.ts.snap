// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader.vue Render MainHeader 1`] = `
<header>
  <div
    class="logo"
  >
    <a
      class="fullName"
      exact=""
      href="/"
    >
       Carlos Hallan 
    </a>
    <span
      class="templateString"
    >
       \`\${
      <span
        class="f"
      >
        User
      </span>
      .
      <span
        class="s"
      >
        name
      </span>
      }\` 
    </span>
  </div>
  <nav>
    
    <a
      class=""
      exact=""
      href="/"
      tag="li"
    >
      Home
    </a>
    <a
      class=""
      exact=""
      href="/work"
      tag="li"
    >
      Work
    </a>
    <a
      class=""
      exact=""
      href="/:slug"
      tag="li"
    >
      post
    </a>
    
  </nav>
</header>
`;
